rpc/wrtc_client_stream.go

```diff
@@ -44,7 +44,7 @@ func newWebRTCClientStream(
 	onDone func(id uint64),
 	logger golog.Logger,
 ) *webrtcClientStream {
-	ctx, cancel := context.WithCancel(ctx)
+	ctx, cancel := utils.MergeContext(channel.ctx, ctx)
```

Comment:
I can't tell this is a behavior change. The specific concern that I'm struggling to reason about:
* If the `ctx` (input parameter from the client/application code) is canceled, that results in [this cancel](https://github.com/viamrobotics/goutils/blob/45e9b67aa447d1d1d09aaf6390c4507b6ab44886/context.go#L48) being called.
* That cancel comes from this [construction of](https://github.com/viamrobotics/goutils/blob/45e9b67aa447d1d1d09aaf6390c4507b6ab44886/context.go#L13) `context.WithCancel(channel.ctx)`.
* Will that result in the `channel.ctx` being canceled?

It's fine if a channel context cancel propagates to streams/API calls (we presume the connection is dead). But it's not ok for a stream/API timeout to break the channel.

Reply:
> Will that result in the channel.ctx being canceled?

Fair concern; I think the answer is no. The code of merged contexts is tough to follow, and I'm getting confused trying to explain my understanding with prose haha. Let me write you a test locally to try to prove that when merging contexts, the first context parameter cannot be canceled by canceling the second or by canceling the merged context.

Reply:
So this test passes locally:

```
func TestMergedContext(t *testing.T) {
	channelCtx, cancelChannelCtx := context.WithCancel(context.Background())
	streamCtx, cancelStreamCtx := context.WithCancel(context.Background())
	mergedCtx, cancelMergedCtx := utils.MergeContext(channelCtx, streamCtx)

	cancelMergedCtx()
	test.That(t, mergedCtx.Err(), test.ShouldNotBeNil)
	test.That(t, mergedCtx.Err(), test.ShouldBeError, context.Canceled)
	test.That(t, streamCtx.Err(), test.ShouldBeNil)
	test.That(t, channelCtx.Err(), test.ShouldBeNil)

	cancelStreamCtx()
	test.That(t, mergedCtx.Err(), test.ShouldNotBeNil)
	test.That(t, mergedCtx.Err(), test.ShouldBeError, context.Canceled)
	test.That(t, streamCtx.Err(), test.ShouldNotBeNil)
	test.That(t, streamCtx.Err(), test.ShouldBeError, context.Canceled)
	test.That(t, channelCtx.Err(), test.ShouldBeNil)

	cancelChannelCtx()
	test.That(t, mergedCtx.Err(), test.ShouldNotBeNil)
	test.That(t, mergedCtx.Err(), test.ShouldBeError, context.Canceled)
	test.That(t, streamCtx.Err(), test.ShouldNotBeNil)
	test.That(t, streamCtx.Err(), test.ShouldBeError, context.Canceled)
	test.That(t, channelCtx.Err(), test.ShouldNotBeNil)
	test.That(t, channelCtx.Err(), test.ShouldBeError, context.Canceled)
}
```
I'm hoping that explains the relationships between the first, second and merged contexts of `MergeContext`. Context erroring propagates as first -> second -> merged.

Reply:
Cool, thanks for checking!